lib/analyzer/applications/node.ts

```diff
@@ -0,0 +1,116 @@
+import * as path from "path";
+import * as lockFileParser from "snyk-nodejs-lockfile-parser";
+
+import * as types from "../../types";
+
+const TYPE = "node";
+const VERSION = "1.0.0";
+
+export interface NodeScanResult extends types.ScanResult {
+  type: string;
+  version: string;
+  data: NodeData;
+}
+
+interface NodeData {
+  parserResult: lockFileParser.PkgTree;
+  lockPath: string;
+  manifestPath: string;
+}
+
+interface FilePathToContent {
+  [filePath: string]: string;
+}
+
+interface ManifestLockPathPair {
+  manifest: string;
+  lock: string;
+  lockType: lockFileParser.LockfileType;
+}
+
+export async function nodeLockFilesToData(
+  filePathToContent: FilePathToContent,
+): Promise<NodeScanResult[]> {
+  const scanResults: NodeScanResult[] = [];
+
+  const filePairs = findManifestLockPairsInSameDirectory(filePathToContent);
+  for (const thing of filePairs) {
+    // TODO: consider spinning the event loop
+    const parserResult = await lockFileParser.buildDepTree(
```

Comment:
what happens if for some reason `buildDepTree` fails? does it throw? just wondering if we need to wrap in try-catch to continue to other files

Reply:
no idea, but good point